AEPIdentity/Sources/Identity.swift

```diff
@@ -224,4 +224,9 @@ import Foundation
 
         return privacyStatus == .optedOut
     }
+
+    /// - Returns: Returns true if the Edge extension is registered, false otherwise
+    private func isEdgeRegistered() -> Bool {
+        return isExtensionRegistered(extensionName: IdentityConstants.SharedStateKeys.EDGE)
```

Comment:
A little concerned about calling this for each hit. Maybe we can use a "local static var" to reduce the number of times this invokes EventHub APIs.

```swift
    private func isEdgeRegistered() -> Bool {
       struct Holder {
           static var edgeRegistered
        }

       if Holder.edgeRegistered { return true }
       Holder.edgeRegistered = isExtensionRegistered(extensionName: IdentityConstants.SharedStateKeys.EDGE)
       return Holder.edgeRegistered
}
```

thoughts?

Reply:
Ideally all the extensions should be registered before start so if we do that for the first event the Edge extension should be registered by that time. 
In addition to that, for subsequent launches we can check if the IdentityMap exists which would indicate the migration happened already once and we don't need to check the edge extension again.
We will not support downgrades to Identity direct once the migration happened.

Reply:
I'm not quite sure I understand the feedback here. This PR specifically adds functionality to disable Identity from sending hits directly to the Identity service if Edge is present. I'd prefer not to have more logic in here than needed with checking if Edge is present, then keeping track of if a migration has occurred. 

It appears to me checking if Edge is present is sufficient enough to determine if Identity should queue/not queue hits. Thoughts?

Reply:
This approach does not cover the use-case where the migration happened, but in the next app launch the Edge extension is not registered anymore. In this case we should not fallback to the Identity direct path.

Reply:
In the case that the migration has happened but Edge is not registered, what would be the behavior of the Identity extension? Do we fallback to not doing anything? When would this scenario occur?

Reply:
Another thought I had is that even if we save a flag or check for the identityMap, it doesn't cover the case where the user uninstalls the app.
1. App launch with Identity + Edge
2. Uninstall the app
3. App launch with only Identity, now we can't determine if the migration happened so we'll fall back on the legacy Identity service, which could lead to inconsistent SDK behavior based on if the user has installed/re-installed. Should this new install be using the Identity service while other users could be going through Edge?

I think that changing the behavior of an extension based on the prior existence of an extension could be error-prone and lead to strange/inconsistent behavior. Unless there is a common workflow where the customer is going to use Edge then remove Edge, I'm not sure we should handle this case.

Reply:
@emdobrin any more thoughts on this?

Reply:
The case where the Edge extension is not registered can occur on app update, maybe the Launch Edge extension is uninstalled or the registration code is removed for any reason. In this case we should update the identifiers in the IdentityMap, but they will not be synced with any service until the Edge extension is registered again.
For the app uninstall scenario you described above, I don't think we can catch that, but we should start thinking about a better representation of this dependency in Launch UI.